### PR TITLE
Return revision metadata instead of just revisionKey

### LINKED ISSUE #14: Create revision metadata instead of just revision key
After discussions on slack, we decided that it would be beneficial to push a json object of metadata to the likes of redis.  It makes sense for this plugin to create that metatdata object, with the revision key being one piece of the metadata, albeit the most important one.

Other data that could be added is the deployer, timestamp, git commit message etc etc.

It might also make sense to rename this repo to something more indicative of it's intention, maybe `ember-cli-deploy-revision-data`.
- [x] Rename repo
- [x] Implement new `prepare` hook

### PR DESCRIPTION
**Pull request by achambers**
This PR slightly changes how this plugin works. Previous to this PR, the plugin would simply return a revision key which would be merged into the deployment context object.

Now, with this PR, the plugin returns an object which can contain any arbitrary meta data about the deploy revision. One proviso is that it must contain a property called `revisionKey`.

The following PR's need to be merged once this one has been:
- [x] https://github.com/zapnito/ember-cli-deploy-redis/pull/27
- [x] https://github.com/lukemelia/ember-cli-deploy-lightning-pack/pull/4
- [x] https://github.com/LevelbossMike/ember-cli-deploy-s3-index/pull/1
- [x] https://github.com/dschmidt/ember-cli-deploy-cp/pull/2

Closes #14

### CODE REVIEW

**Comment by achambers** [comment]
@LevelbossMike, you will need to make a change to your ember-cli-deploy-s3-index repo to come in to line with this PR.

File: README.md [hunk 1]
```diff
@@ -51,18 +52,28 @@ For detailed information on how configuration of plugins works, please refer to
...
+The type of [Data Generator](#data-generators) to be used.
 
 *Default:* `'file-hash'`
 *Alternatives:* `'git-tag-commit'`, `'version-commit'`
 
-## Key Generators
+## Data Generators
 
-Key generators are the strategies used to generate the unique revision key. Currently there is only one Key Generator implementation but we will add more as needed. Some examples of other potential key generators are `GitHash` or `DateTime` generators.
+Data generators are the strategies used to generate information about the revision being deployed. A data generator must return an object which contains a property called `revisionKey` which uniquely identifies the current revision. A generator can add any other data that it deems relevant to the data object that it returns.
```
Comment (lukemelia) [inline README.md:62]: README reader may want to know: how do I specify my own data generator?

Comment (lukemelia) [inline README.md:62]: Related question that comes to mind: should we have one plugin with separate data generators? Or separate plugins?

Comment (achambers) [inline README.md:62]: > README reader may want to know: how do I specify my own data generator?

Good point. Will add something to this effect

> Related question that comes to mind: should we have one plugin with separate data generators? Or separate plugins?

Yeh, this is something I've been wondering today actually. Was talking to a mate who indicated a need for a different data generates which got me asking myself the same questions you have.

I'm starting to think maybe separate plugins could be the way forward. It makes them a bit disparate but maybe it makes more sense than adding a large list of generators that are probably going to be for people's specific use cases initially.

Either way it feels like a separate piece of work than what this PR is intended for. But yep, I'm starting to lean down the separate plugin route myself.

Comment (lukemelia) [inline README.md:62]: Merging this then, and we can proceed with splitting it up in the future.

Comment (lukemelia) [inline README.md:62]: Actually, will wait for your README tweak.

Comment (achambers) [inline README.md:62]: Yep, will do the README change first.

**Pull request merged**